### PR TITLE
fleetctl: sanity check unit names in submit/load/start and disallow templates in load/start

### PR DESCRIPTION
**Pull request by mischief**

### CODE REVIEW

File: fleetctl/fleetctl.go [hunk 1]
```diff
@@ -845,3 +845,28 @@ func suToGlobal(su schema.Unit) bool {
 	}
 	return u.IsGlobal()
 }
+
+// checkUnitNames sanity checks unit names
+func checkUnitNames(names []string, allowTemplate bool) error {
+	units := make([]string, 0, len(names))
```
Comment (mwhooker) [inline fleetctl/fleetctl.go]: forgive me if I'm misunderstanding the intention, but I think this should be rewritten as 

```go
    func checkUnitNames(names []string, allowTemplate bool) error {
	for _, un := range units {
		uni := unit.NewUnitNameInfo(unitNameMangle(un))
		// snip...
```

Comment (mischief) [inline fleetctl/fleetctl.go]: sounds good. my original intention was to drop unit names that failed validation from the slice, but i feel like it's better to abort if any unit name fails checks.

unfortunately the combination of `unit.NewUnitNameInfo` and `unitNameMangle` is not terribly helpful, becuase `unit.NewUnitNameInfo` only returns nil if the unit is missing a `.` and `unitNameMangle` is appending `.service` for us. more checks are probably needed in `unit.NewUnitNameInfo`.

Comment (mwhooker) [inline fleetctl/fleetctl.go]: yeah, might be best to write some tests that demonstrate which names fail the check and which succeed

**Comment by mischief** [comment]
closing in favor of checking for this in the fleet api.

**Pull request closed without merging**